workflows/background_check.go

```diff
@@ -0,0 +1,192 @@
+package workflows
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/temporalio/background-checks/config"
+	"github.com/temporalio/background-checks/mappings"
+	"github.com/temporalio/background-checks/queries"
+	"github.com/temporalio/background-checks/types"
+	"go.temporal.io/sdk/log"
+	"go.temporal.io/sdk/workflow"
+)
+
+type backgroundCheckWorkflow struct {
+	types.BackgroundCheckState
+	checkID       string
+	ctx           workflow.Context
+	checkFutures  []workflow.Future
+	checkSelector workflow.Selector
+	logger        log.Logger
+}
+
+func newBackgroundCheckWorkflow(ctx workflow.Context, state types.BackgroundCheckState) (*backgroundCheckWorkflow, error) {
+	w := backgroundCheckWorkflow{
+		BackgroundCheckState: state,
+		checkID:              workflow.GetInfo(ctx).WorkflowExecution.RunID,
+		ctx:                  ctx,
+		checkSelector:        workflow.NewSelector(ctx),
+		logger:               workflow.GetLogger(ctx),
+	}
+
+	w.Checks = make(map[string]interface{})
+
+	err := workflow.SetQueryHandler(ctx, queries.BackgroundCheckStatus, func() (types.BackgroundCheckState, error) {
+		return w.BackgroundCheckState, nil
+	})
+	return &w, err
+}
+
+func (w *backgroundCheckWorkflow) pushStatus(status types.BackgroundCheckStatus) error {
+	return workflow.UpsertSearchAttributes(
+		w.ctx,
+		map[string]interface{}{
+			"BackgroundCheckStatus": status.String(),
+		},
+	)
+}
+
+func (w *backgroundCheckWorkflow) waitForAccept(email string) (types.AcceptSubmission, error) {
+	var r types.AcceptSubmission
+
+	err := w.pushStatus(types.BackgroundCheckStatusPendingAccept)
+	if err != nil {
+		return r, err
+	}
+
+	checkID := workflow.GetInfo(w.ctx).WorkflowExecution.RunID
+
+	ctx := workflow.WithChildOptions(w.ctx, workflow.ChildWorkflowOptions{
+		WorkflowID: mappings.AcceptWorkflowID(checkID),
+	})
+	consentWF := workflow.ExecuteChildWorkflow(ctx, Accept, types.AcceptWorkflowInput{
+		Email:   email,
+		CheckID: checkID,
+	})
+	err = consentWF.Get(ctx, &r)
```

Comment:
If you're just executing a child workflow and then synchronously waiting on the result always, what benefit is there in using a child workflow?

Reply:
Discussed elsewhere.